lua/lsp-format/init.lua

```diff
@@ -80,6 +80,21 @@ local parse_value = function(key, value)
     return value
 end
 
+---@param bufnr number
+---@param options lsp.FormattingOptions
+---@return lsp.DocumentFormattingParams
+local function make_formatting_params(bufnr, options)
+    local tabSize = vim.api.nvim_get_option_value("shiftwidth", { buf = bufnr })
+    local expandtab = vim.api.nvim_get_option_value("expandtab", { buf = bufnr })
+    if expandtab or tabSize == 0 then
+        tabSize = vim.api.nvim_get_option_value("tabstop", { buf = bufnr })
+    end
```

Comment:
Why is this logic different from the one used earlier (https://github.com/neovim/neovim/blob/e268fcbdaa1e0e0cee3b513e62581d35bb937d40/runtime/lua/vim/lsp/util.lua#L2078-L2083)? Can't we just use `get_effective_tabstop` directly?

Specifically, `vim.lsp.util.make_formatting_params` will use `sw` even if `expandtab`, but your implementation will use `tabstop` instead in such case.